src/db/spanner/models.rs

```diff
@@ -618,21 +622,21 @@ impl SpannerDb {
             .param_types(param_types! {
                 "pretouch_ts" => TypeCode::TIMESTAMP,
             })
-            .execute(&self.conn)?
-            .map_and_then(|row| {
-                let collection_id = row[0]
-                    .get_string_value()
-                    .parse::<i32>()
-                    .map_err(|e| DbErrorKind::Integrity(e.to_string()))?;
-                let modified = SyncTimestamp::from_rfc3339(&row[1].get_string_value())?;
-                Ok((collection_id, modified))
-            })
-            .collect::<Result<HashMap<_, _>>>()?;
-        self.map_collection_names(modifieds)
+            .execute_async(&self.conn)?;
+        let mut results = HashMap::new();
+        while let Some(row) = streaming.one_or_none().await? {
```

Comment:
These iterating blocks shouldn't use `one`/`one_or_none` (they error out unless there's *only* one result), did db-tests not catch this (they really should)?

You probably just want streaming.next_async().await.transpose()? (the need to transpose is a little annoying, I wonder what it would take to implement the `Stream` type for this).

Reply:
I guess the async version doesn't error out if there is more than one, which is why it works. Shall I rename it next_or_none?

Reply:
Huh, no, you're right, it should error if there is more than one. Weird. Yes it seems like the db tests should be catching this.

Reply:
I guess they can just use next_async.